### DISCUSSION TITLE
input datetime e template a cavallo della mezzanotte

### DISCUSSION THREAD

**Discussion by serlinGi** [indomus/forum#640] [GENERALE - Discussioni generiche]
Ciao a tutti;
da tempo gestisco alcune automazioni grazie ai vari suggerimenti vostri ed in particolare di @snakuzzo  per esecuzioni basate sugli orari impostati con l'input_datetime con i template;
mi stavo giusto stamattina accingendo ad impostare dei range orari nelle condizioni
tipo questo  

https://pastebin.com/FcSG9eHq

ma mi sono reso conto che se l'orario di spegnimento è in valore assoluto inferiore a quello di accensione, la cosa non può andare;
per spiegarmi meglio:
se ON->09:00; OFF->16:00 funzionerebbe 
ma se invece ON->16:00 ; OFF->09:00 
la cosa dovrebbe riuscire "male" !

Suggerimenti ulteriori? 
:D

**Comment by Larry-78** [comment]
Ciao, dovresti utilizzare una Time condition con entrambi `after:` e `before:` per poter andare a cavallo della mezzanotte.
Il riferimento alla documentazione ufficiale lo trovi [qui](https://www.home-assistant.io/docs/scripts/conditions#time-condition) con eventuali esempi.

**Reply by serlinGi** [reply]
Grazie mille;
è il tipico esempio di  quando le cose semplici surclassano le antiche abitudini;
uso in altre situazioni il platform time con after e before ma non avevo idea si potesse applicare ad un input_datetime... 
grazie ;)

**Reply by Larry-78** [reply]
Di nulla 😄👍🏻
